bootstrap/bootstrap.go

```diff
@@ -424,8 +426,18 @@ func (b *Bootstrap) setUp() error {
 		if b.BuildPath == "" {
 			return fmt.Errorf("Must set either a BUILDKITE_BUILD_PATH or a BUILDKITE_BUILD_CHECKOUT_PATH")
 		}
-		b.shell.Env.Set("BUILDKITE_BUILD_CHECKOUT_PATH",
-			filepath.Join(b.BuildPath, dirForAgentName(b.AgentName), b.OrganizationSlug, b.PipelineSlug))
+
+		// If we should consolidate repos, hash the repository path to use as the build dir
+		if b.Config.ShouldConsolidateRepos && b.Config.Repository != "" {
+
+			hashedRepositoryName := hashRepositoryString(b.Config.Repository)
+
+			b.shell.Env.Set("BUILDKITE_BUILD_CHECKOUT_PATH",
```

Comment:
Is this mutually exclusive with a node running more than one agent at a time?

Reply:
Yes, `b.shell.Env.Set("BUILDKITE_BUILD_CHECKOUT_PATH")` is used in other contexts and is exclusive to the current shell